### ISSUE TITLE
Option for lens definition in submodule

### ISSUE THREAD

**Issue by ghost** [pdonadeo/ocaml-lens#5]
Could there be an option like `prefix`, but instead it creates a `Foo.Lens` sub-module?

With that feature, one could use the lens function without function name noise or shadowing the record identifiers.

```
module Foo = struct
  type t = { foo : f_t; bar : bar_t } [@@deriving lens { module = true }]
end

let get_foo a = Foo.Lens.( a |. foo)
```

**Comment by pdonadeo** [comment]
@didier-wenzek do you think you have time to add this feature?

**Comment by didier-wenzek** [comment]
This is a nice idea. I'll take the time to make a pull request this week.

**Comment by pdonadeo** [comment]
@didier-wenzek thanks. I'm not familiar with ppx_deriving, so I'm waiting for your pull request to merge and release.

**Comment by ghost** [comment]
@pdonadeo same thing here with ppx_deriving.

**Comment by didier-wenzek** [comment]
On my side, I have trouble with oasis :-(.
I failed to launch ppx_test. The dependency to `lens.ppx_deriving` is missing on `make test` while correctly requested in `_oasis`. This is not blocking. I have a workaround using directly `ocamlbuild`.

For ppx_deriving, a must have are the tools of https://github.com/alainfrisch/ppx_tools. For instance, I use `ocamlfind ppx_tools/dumpast` to generate the AST shape of the code to derive.

**Comment by ghost** [comment]
@didier-wenzek did you get past the dependency issues?

**Comment by didier-wenzek** [comment]
I managed to code & test the feature using a workaround for the tests (I compiled the tests using a hand-written `_tags` file). This is not compatible with oasis and a bit convoluted (Hence, I did not add these changes in the pull request).

Beyond this issue, things went fine.
I submitted a pull request https://github.com/pdonadeo/ocaml-lens/pull/6.

I changed the name of the option to `submodule = true` since using the term `module` leads to syntax errors.
